sdk/schemaregistry/azure-schemaregistry-avroencoder/azure/schemaregistry/encoder/avroencoder/aio/_schema_registry_avro_encoder_async.py

```diff
@@ -264,36 +310,78 @@ async def decode(
             try:
                 content = message["content"]
                 content_type = message["content_type"]
-            except (KeyError, TypeError):
-                SchemaDecodeError(
-                    f"""The content model {str(message)} is not a subtype of the MessageType protocol or type
-                        MessageContent.  If using an Azure SDK model class, please check the README.md
+            except (KeyError, TypeError) as exc:
+                raise TypeError(
+                    f"""The content model {str(message)} must be a subtype of the MessageType protocol or type
+                        MessageContent. If using an Azure SDK model class, please check the README.md
                         for the full list of supported Azure SDK models and their corresponding versions."""
-                ).raise_with_traceback()
+                ) from exc
 
-        # include in first preview for back compatibility
-        content, content_type = self._convert_preamble_format(content, content_type)
-
-        schema_id = content_type.split("+")[1]
+        try:
+            schema_id = content_type.split("+")[1]
+        except AttributeError:
+            raise InvalidContentError(
+                "Content type was not in the expected format of MIME type + schema ID."
+            )
 
-        cache_misses = self._get_schema.cache_info().misses # pylint: disable=no-value-for-parameter disable=no-member
+        cache_misses = (
+            self._get_schema.cache_info().misses    # pylint: disable=no-value-for-parameter disable=no-member
+        )
         request_options = request_options or {}
         schema_definition = await self._get_schema(schema_id, **request_options)
-        new_cache_misses = self._get_schema.cache_info().misses # pylint: disable=no-value-for-parameter disable=no-member
+        new_cache_misses = (
+            self._get_schema.cache_info().misses    # pylint: disable=no-value-for-parameter disable=no-member
+        )
         if new_cache_misses > cache_misses:
-            cache_size = self._get_schema.cache_info().currsize  # pylint: disable=no-value-for-parameter disable=no-member
-            _LOGGER.info("New entry has been added to schema cache. Cache size: %s", str(cache_size))
+            cache_info = (
+                self._get_schema.cache_info()  # pylint: disable=no-value-for-parameter disable=no-member
+            )
+            _LOGGER.info(
+                "New entry has been added to schema cache. Cache info: %s",
+                str(cache_info),
+            )
+
+        try:
```

Comment:
the following decode code is also the same as sync one right? (as there's no async apach avro decoder..)
we could de-dup the codes as well